### ISSUE TITLE
Bundler disabling Compact Index on all FIPS-supported OpenSSL installations

### ISSUE THREAD

**Issue by wjordan** [bundler/bundler#5433]
Since #5222, Bundler is disabling Compact Index on any system where OpenSSL is compiled with _support_ for FIPS mode, regardless of whether or not FIPS mode is currently _enabled_ on the process (e.g., by setting `OpenSSL.fips_mode = true` in Ruby).

Compare the following:

FIPS mode supported but not enabled:

```
$ ruby -ropenssl -rdigest/md5 -e "p OpenSSL::OPENSSL_FIPS; p Digest::MD5.hexdigest('foo')"
true
"acbd18db4cc2f85cedef654fccc4a4d8"
```

FIPS mode supported and enabled:

```
$ ruby -ropenssl -rdigest/md5 -e "p OpenSSL::OPENSSL_FIPS; OpenSSL.fips_mode = true; p Digest::MD5.hexdigest('foo')"
true
md5_dgst.c(83): OpenSSL internal error, assertion failed: Digest MD5 forbidden in FIPS mode!
Aborted
```

This bug is affecting current versions of Amazon Linux, which have `OpenSSL 1.0.1k-fips` installed by default.

**Comment by segiddins** [comment]
A PR with a fix for this would be much appreciated!

**Comment by wjordan** [comment]
PR #5440 is the best I can do to extend #5222's detection-based approach to the underlying FIPS compatibility issue (#4989), but I don't think it's a good approach. I think a better fix for this issue is to revert #5222 and come up with a better solution to support FIPS.
